// @dcloudio/uni-h5-vite/dist/plugin/transformIndexHtml/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTransformIndexHtml = void 0;
const uni_cli_shared_1 = require("@dcloudio/uni-cli-shared");
function createTransformIndexHtml() {
    return async function (html) {
        var _a;
        const manifestJson = (0, uni_cli_shared_1.parseManifestJsonOnce)(process.env.UNI_INPUT_DIR);
        const title = ((_a = manifestJson.h5) === null || _a === void 0 ? void 0 : _a.title) || manifestJson.name || '';
        return {
            html: html.replace(/<title>(.*?)<\/title>/, `<title>${title}</title>`),
            tags: process.env.NODE_ENV === 'development'
                ? [
                    {
                        tag: 'script',
                        children: `if (typeof globalThis === 'undefined') {
  window.globalThis = window
}`,
                        injectTo: 'head-prepend',
                    },
                ]
                : [],
        };
    };
}
exports.createTransformIndexHtml = createTransformIndexHtml;
